pkg/cmd/server/kubernetes/node.go

```diff
@@ -140,20 +168,24 @@ func (c *NodeConfig) RunProxy() {
 	// initialize kube proxy
 	serviceConfig := pconfig.NewServiceConfig()
 	endpointsConfig := pconfig.NewEndpointsConfig()
-	pconfig.NewConfigSourceEtcd(c.EtcdClient,
-		serviceConfig.Channel("etcd"),
-		endpointsConfig.Channel("etcd"))
+	pconfig.NewSourceAPI(
+		c.Client.Services(kapi.NamespaceAll),
+		c.Client.Endpoints(kapi.NamespaceAll),
+		30*time.Second,
+		serviceConfig.Channel("api"),
+		endpointsConfig.Channel("api"))
 	loadBalancer := proxy.NewLoadBalancerRR()
 	endpointsConfig.RegisterHandler(loadBalancer)
 
-	// TODO clearly this needs fixing
+	ip := net.ParseIP(c.BindHost)
```

Comment:
huh, I guess it would have died before as well... a returned error would be far more helpful